internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -241,6 +256,91 @@ func (r *ClusterDeploymentsReconciler) Reconcile(origCtx context.Context, req ct
 	return r.updateStatus(ctx, log, clusterInstall, cluster, nil)
 }
 
+func (r *ClusterDeploymentsReconciler) setHTTPSIgnitionURL(ctx context.Context, log logrus.FieldLogger, cluster *common.Cluster, clusterInstall *hiveext.AgentClusterInstall, secretRef types.NamespacedName) error {
+
+	if clusterInstall.Spec.IgnitionEndpoint == nil {
+		log.Debugf("Empty endpoint found")
+		// Set custom Ignition endpoint so that day2 workers would join using HTTPS endpoint
+		// ensureMCSCert would add ignition override with MCS secret
+		ignitionEndpoint := fmt.Sprintf("%s/config/%s", common.GetMCSUrlBase(cluster), models.HostRoleWorker)
+		log.Infof("setting worker ignition URL to %s", ignitionEndpoint)
+		clusterInstall.Spec.IgnitionEndpoint = &hiveext.IgnitionEndpoint{
+			Url: ignitionEndpoint,
+			CaCertificateReference: &hiveext.CaCertificateReference{
+				Name:      secretRef.Name,
+				Namespace: secretRef.Namespace,
+			},
+		}
+		if err := r.Update(ctx, clusterInstall); err != nil {
+			return err
+		}
+		log.Debugf("Endpoint updated")
+	}
+	return nil
+}
+
+func (r *ClusterDeploymentsReconciler) createMCSCertSecret(ctx context.Context, log logrus.FieldLogger, clusterDeployment *hivev1.ClusterDeployment, cluster *common.Cluster, clusterInstall *hiveext.AgentClusterInstall) (types.NamespacedName, error) {
+
```

Comment:
Consider refactoring it to:
GetSpokeClusterClient (pretty sure it's already implemented)
GetMCSCertInfo